zstacklib/zstacklib/utils/gpu.py

```
@@ -233,3 +234,40 @@ def is_hygon_gpu_cmd(pci_addr):
+        return None
+    json_str = output[json_start:]
+    try:
+        return json.loads(json_str)
+    except ValueError as ve:
+        logger.error("JSON decode failed for command: %s, error: %s", cmd, ve)
+        return None
+
+
+def run_command_with_realtime_output(command, timeout=30):
+    """Run a command and yield its output line by line."""
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Ruff (0.8.2)

`subprocess` may be undefined, or defined from star imports

(F405)